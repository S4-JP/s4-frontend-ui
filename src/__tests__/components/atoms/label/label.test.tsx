import React from 'react'
import { render } from '@testing-library/react'
import { LabelEmail, LabelIcon, LabelSeverity, Color } from '../../../../index'

describe('label - LabelEmail のレンダリング確認', () => {
  test('オプションなしで設定した場合、初期値でレンダリングされる。', () => {
    const { getByTestId } = render(<LabelEmail label="test" />)

    expect(getByTestId('atoms-l-label-email-wrap')).toMatchSnapshot()
  })
})

describe('label - LabelIcon のレンダリング確認', () => {
  test('オプションなしで設定した場合、初期値でレンダリングされる。', () => {
    const { getByTestId } = render(<LabelIcon label="未対応" />)

    expect(getByTestId('atoms-l-label-icon-text')).toMatchSnapshot()
  })
})

type Value = {
  label: string
  expect: string
}
const values: Value[] = [
  { label: '未対応', expect: 'atoms-icon-svg-status-backlog' },
  { label: '対応中', expect: 'atoms-icon-svg-status-progress' },
  { label: '対応済み', expect: 'atoms-icon-svg-status-done' },
  { label: '対応なし', expect: 'atoms-icon-svg-status-closed' },
]

describe.each(values)('LabelIcon - labelを設定した場合', (v) => {
  test(`value:「${v.label}」の時、設定したアイコン「${v.expect}」が表示される。`, () => {
    const { getByTestId } = render(<LabelIcon label={v.label} />)
    expect(getByTestId(v.expect)).toBeInTheDocument()
  })
})

/**
 * type Props = {
 *   severityId: number
 *   severity: string
 *   score: number
 *   version?: number
 *   small?: boolean
 *   short?: boolean
 * }
 */
describe('label - LabelSeverity のレンダリング確認', () => {
  test('オプションなしで設定した場合、初期値でレンダリングされる。', () => {
    const { getByTestId } = render(
      <LabelSeverity severityId={1} severity="緊急" score={10.0} />
    )

    expect(getByTestId('atoms-l-label-severity-wrap')).toMatchSnapshot()
  })
})

type SevValue = {
  severityId: number
  severity: string
  score: number
  expectBg: string
  expectTc: string
}
const sevValues: SevValue[] = [
  {
    severityId: 5,
    severity: '緊急',
    score: 10.0,
    expectBg: Color.SEVERITY.CRITICAL,
    expectTc: Color.SEVERITY.CRITICAL,
  },
  {
    severityId: 4,
    severity: '高',
    score: 8.0,
    expectBg: Color.SEVERITY.HIGH,
    expectTc: Color.SEVERITY.HIGH,
  },
  {
    severityId: 3,
    severity: '中',
    score: 6.0,
    expectBg: Color.SEVERITY.MEDIUM,
    expectTc: Color.TEXT.GRAY,
  },
  {
    severityId: 2,
    severity: '低',
    score: 3.0,
    expectBg: Color.SEVERITY.LOW,
    expectTc: Color.TEXT.GRAY,
  },
  {
    severityId: 1,
    severity: 'なし',
    score: 0.0,
    expectBg: Color.SEVERITY.INFO,
    expectTc: Color.TEXT.GRAY,
  },
]

describe.each(sevValues)('LabelSeverity - labelを設定した場合', (v) => {
  test(`value:「${v.severity}」の時、それぞれ設定される。`, () => {
    const { getByTestId } = render(
      <LabelSeverity
        severityId={v.severityId}
        severity={v.severity}
        score={v.score}
      />
    )

    expect(getByTestId('atoms-l-label-severity-urgency-bar')).toHaveStyle(
      `background: ${v.expectBg}`
    )
    expect(getByTestId('atoms-l-label-severity-severity-text')).toHaveStyle(
      `color: ${v.expectTc}`
    )
    expect(getByTestId('atoms-l-label-severity-score').textContent).toBe(
      '' + v.score
    )
  })
})

describe('label - LabelSeverity オプションのレンダリング確認', () => {
  test('props.small = true で設定した場合、設定値でレンダリングされる。', () => {
    const { getByTestId } = render(
      <LabelSeverity severityId={1} severity="緊急" score={10.0} small={true} />
    )

    expect(getByTestId('atoms-l-label-severity-score')).toHaveStyle(
      'font-size: 18px;'
    )
    expect(getByTestId('atoms-l-label-severity-urgency-bar')).toHaveStyle(
      'width: 8px; height: 48px;margin: 0 8px 0 0; border-radius: 14px'
    )
  })
})
