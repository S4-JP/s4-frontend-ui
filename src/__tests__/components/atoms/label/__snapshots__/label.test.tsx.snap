// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`label - LabelEmail のレンダリング確認 オプションなしで設定した場合、初期値でレンダリングされる。 1`] = `
<div
  class="sc-lbxAil cZMthh"
  data-testid="atoms-l-label-email-wrap"
>
  <div
    class="sc-iNWwEs eyQrRp"
    data-testid="atoms-l-label-email-text-wrap"
  >
    test
  </div>
  <div
    class="sc-jfmDQi foLnOQ"
    data-testid="atoms-l-label-email-icon-wrap"
  >
    <span
      class="sc-papXJ ZErXk"
      color="#92A2AF"
      data-testid="atoms-i-icon-wrap"
      size="16"
    >
      <svg
        data-testid="atoms-icon-svg-circle-cross"
        viewBox="0 0 40 40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20.001-.003a20 20 0 1 0 20 20 20 20 0 0 0-20-20Zm9.17 26.173a2.121 2.121 0 1 1-3 3l-6.174-6.174-6.174 6.174a2.169 2.169 0 0 1-3 0 2.123 2.123 0 0 1 0-3l6.172-6.172-6.172-6.174a2.121 2.121 0 1 1 3-3l6.174 6.174 6.174-6.174a2.121 2.121 0 1 1 3 3l-6.172 6.174Z"
        />
        <path
          d="M0 0h40v40H0z"
          fill="none"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`label - LabelIcon のレンダリング確認 オプションなしで設定した場合、初期値でレンダリングされる。 1`] = `
<div
  class="sc-cTQhss fIxAuC"
  data-testid="atoms-l-label-icon-text"
>
  <div
    class="sc-jOrMOR ddwLxI"
    data-testid="atoms-l-label-icon-wrap"
  >
    <span
      class="sc-papXJ cBqjuY"
      data-testid="atoms-i-icon-wrap"
    >
      <svg
        data-testid="atoms-icon-svg-status-backlog"
        viewBox="0 0 40 40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h40v40H0z"
          fill="none"
        />
        <g
          fill="#ffa5a5"
        >
          <path
            d="M7.412 15.798a13.056 13.056 0 0 1 .705-1.7 3.763 3.763 0 0 0-.746-4.325 3.823 3.823 0 0 0-6.122.977 20.976 20.976 0 0 0-1.04 2.5 3.83 3.83 0 0 0 3.627 5.056 3.749 3.749 0 0 0 3.576-2.508ZM14.1 8.115a13.336 13.336 0 0 1 1.7-.7 3.747 3.747 0 0 0 2.5-3.576A3.83 3.83 0 0 0 13.246.212a21 21 0 0 0-2.5 1.04 3.824 3.824 0 0 0-.975 6.122 3.766 3.766 0 0 0 4.329.741ZM8.117 25.9a12.86 12.86 0 0 1-.705-1.7 3.75 3.75 0 0 0-3.578-2.5 3.828 3.828 0 0 0-3.627 5.054 21.118 21.118 0 0 0 1.04 2.5 3.823 3.823 0 0 0 6.122.975 3.765 3.765 0 0 0 .748-4.329ZM21.698 3.833a3.749 3.749 0 0 0 2.5 3.578 13.288 13.288 0 0 1 1.7.7 3.763 3.763 0 0 0 4.325-.746 3.823 3.823 0 0 0-.977-6.122 21 21 0 0 0-2.5-1.04 3.83 3.83 0 0 0-5.048 3.63ZM31.884 14.099a13.088 13.088 0 0 1 .7 1.7 3.749 3.749 0 0 0 3.578 2.5 3.83 3.83 0 0 0 3.629-5.054 20.614 20.614 0 0 0-1.041-2.5 3.821 3.821 0 0 0-6.121-.977 3.76 3.76 0 0 0-.745 4.331ZM32.588 24.201a13.21 13.21 0 0 1-.7 1.7 3.76 3.76 0 0 0 .744 4.325 3.821 3.821 0 0 0 6.121-.977 20.841 20.841 0 0 0 1.041-2.5 3.829 3.829 0 0 0-3.629-5.054 3.75 3.75 0 0 0-3.577 2.506ZM25.901 31.885a13.27 13.27 0 0 1-1.7.7 3.749 3.749 0 0 0-2.5 3.578 3.829 3.829 0 0 0 5.054 3.627 20.44 20.44 0 0 0 2.5-1.04 3.822 3.822 0 0 0 .977-6.121 3.76 3.76 0 0 0-4.331-.744ZM15.799 32.585a13.072 13.072 0 0 1-1.7-.7 3.763 3.763 0 0 0-4.327.744 3.822 3.822 0 0 0 .975 6.121 20.711 20.711 0 0 0 2.5 1.04 3.829 3.829 0 0 0 5.054-3.627 3.75 3.75 0 0 0-2.502-3.578Z"
          />
        </g>
      </svg>
    </span>
  </div>
  未対応
</div>
`;

exports[`label - LabelSeverity のレンダリング確認 オプションなしで設定した場合、初期値でレンダリングされる。 1`] = `
<div
  class="sc-dPyBCJ"
  data-testid="atoms-l-label-severity-wrap"
>
  <div
    class="sc-bBXxYQ bfrJJn"
    data-testid="atoms-l-label-severity-bar-wrap"
  >
    <div
      class="sc-gSAPjG bEeAWZ"
      data-testid="atoms-l-label-severity-urgency-bar"
    >
      <span
        class="sc-jgbSNz cnviBD"
        data-testid="atoms-l-label-severity-label-text"
      >
        緊急
      </span>
    </div>
    <div
      class="sc-cOFTSb cshTAr"
      data-testid="atoms-l-label-severity-content-wrap"
    >
      <div
        class="sc-kgUAyh cMYWVd"
        data-testid="atoms-l-label-severity-severity-text"
      >
        <span
          class="sc-hTtwUo eHAjex"
          data-testid="atoms-l-label-severity-version"
        >
          CVSS
        </span>
        v
      </div>
      <div
        class="sc-BeQoi grDrhS"
        data-testid="atoms-l-label-severity-score"
      >
        10
      </div>
    </div>
  </div>
</div>
`;
